File: backend/src/main/java/com/backend/domain/comment/repository/CommentRepository.java
package com.backend.domain.comment.repository;

import com.backend.domain.comment.entity.Comment;
import com.backend.domain.post.entity.Post;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface CommentRepository extends JpaRepository<Comment, Long> {

    @EntityGraph(attributePaths = {"author", "post"})
    Page<Comment> findByPostId(Long postId, Pageable pageable);

    @EntityGraph(attributePaths = {"author", "post"})
    Page<Comment> findByAuthorId(Long authorId, Pageable pageable);

}

File: backend/src/main/java/com/backend/domain/comment/repository/CommentRepository.kt
package com.backend.domain.comment.repository

import com.backend.domain.comment.entity.Comment
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository

interface CommentRepository : JpaRepository<Comment?, Long?> {
    @EntityGraph(attributePaths = ["author", "post"])
    fun findByPostId(postId: Long, pageable: Pageable): Page<Comment>

    @EntityGraph(attributePaths = ["author", "post"])
    fun findByAuthorId(authorId: Long, pageable: Pageable): Page<Comment>
}
